packages/api/src/lib/homework.ts

```diff
@@ -10,10 +11,22 @@ export interface HomeworkInfo {
 
 export interface LearnerHomeworkInfo {
   name: string;
-  date: string;
+  homeworkGivenDate: string;
+  answerReleaseDate: string;
   maxPoints: number;
   myScore: number;
   avgScore: number;
   archive: string;
   filepath: string;
 }
+
+export interface HomeworkData {
```

Comment:
If this is a request type, then please name it so. You can create a new type like so, instead of copy paste:

export type CreateHomeworkRequest = Omit<HomeworkInfo, 'homeworkId'>;

Reply:
For the CreateHomeworkRequest type, we need to make the homeworkId optional, since it's not required when creating a new homework item. However, it is mandatory in the update API.

Rather than omitting homeworkId entirely (which would remove it from the type), we can override the homeworkId property and make it optional while retaining all other fields from the HomeworkInfo interface. 
is it fine?
export type CreateHomeworkRequest = Omit<HomeworkInfo, 'homeworkId'> & {
  homeworkId?: number;
};

Reply:
It is better to create two separate objects. One object for CreateHomeworkRequest and another for UpdateHomeworkRequest

UpdateHomeworkRequest will be same as the original data type

Reply:
Okay !